### PR TITLE
* Feat Added new functionality for MQAP Partner matching AI

### PR DESCRIPTION
**Pull request by JuanGuzman-io**

### CODE REVIEW

**Comment by JuanGuzman-io** [comment]
> @xKeCo @JuanGuzman-io please check Sonar's recommendations.

We are working on it 👍🏻

File: onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-partners/components/knowledge-product-selector/knowledge-product-selector.component.ts [hunk 1]
```diff
@@ -17,12 +17,29 @@ export class KnowledgeProductSelectorComponent {
...
+    const institutionFinded = insts.find(institution => institution.institutions_id == partner.institutions_id);
+
+    partner.obj_institutions.obj_institution_type_code.name = institutionFinded?.institutions_type_name;
+    partner.obj_institutions.website_link = institutionFinded?.website_link;
+  }
+
+  generateDescription(partner) {
+    const confidenceLevel = partner.result_kp_mqap_institution_obj.confidant;
+
+    if (confidenceLevel > 90) {
```
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-partners/components/knowledge-product-selector/knowledge-product-selector.component.ts:39]: isn't the confidence level supposed to be variable? @xKeCo

File: onecgiar-pr-server/src/api/results/results_by_institutions/results_by_institutions.service.ts
Comment (gcmartinezg) [inline onecgiar-pr-server/src/api/results/results_by_institutions/results_by_institutions.service.ts:1]: I am not a fan of calling saves inside of a loop. Feel free to disregard this comment though.

**Review by gcmartinezg — APPROVED** [review]

File: onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-partners/components/knowledge-product-selector/knowledge-product-selector.component.ts (hunk 1 above)
Comment (xKeCo) [reply onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-partners/components/knowledge-product-selector/knowledge-product-selector.component.ts:39]: Yes, that is correct. This development is associated with User Story #2, which is still in progress. Once this activity is completed, this value will become dynamic. @gcmartinezg

**Pull request merged**